lib/Basics/debugging.cpp

```diff
@@ -149,6 +158,7 @@ void TRI_ClearFailurePointsDebugging() noexcept {
     numExisting = ::failurePoints.size();
     ::failurePoints.clear();
   }
+  ::hasFailurePoints.store(false, std::memory_order_relaxed);
```

Comment:
This could also cause problems because we are not updating the atomic variable while under the lock.
Imagine we are removing all failure points here and then are about to store false in the atomic variable.
Another thread could slip in here and a new a failure point before we are setting the atomic variable to `false`. Then this API will unconditionally overwrite the atomic's `true` value with a `false`, and we will have an inconsistent state.

Reply:
Fixed!